docs/development.md

```diff
@@ -67,93 +106,147 @@ docker compose down -v
 
 This should only need to be done once. When the `db` service starts up again (including via running the `load-db` command), the necessary roles and databases will be created automatically.
 
-# Running the server
+<details>
+<summary><b>Don't have a NERSC account?</b></summary>
+If you're an NMDC team member, but don't have a NERSC account yet: talk to your team lead about getting a NERSC account, specifically one that has access to NMDC's project files. The process takes about a week. Docs: https://docs.nersc.gov/accounts/
+
+If that is not an option for you:
+
+1. Ask an NMDC team member with NERSC access to get a recent production backup for you. This will come in the form of a `.dump` file. Save the file locally.
+2. Bring your local database up
+    ```bash
+    docker compose up db -d
+    ```
+3. Load data from the `.dump` file into the running database
+    ```bash
+    docker compose run --rm \
+        -v <absolute path to .dump file>:/tmp/backup.dump \
+       db pg_restore --dbname postgresql://postgres:postgres@host.docker.internal:5432/nmdc_a --clean --if-exists --verbose --single-transaction /tmp/backup.dump
+    ```
+</details>
+
+## Installing dependencies locally
+
+Although the project is designed to be run in Docker, having the dependencies installed locally can be useful for development (e.g. providing code completion in your editor, running tests, etc.).
+
+### Backend dependencies
+
+1. If necessary, create a new virtual environment.
+    ```bash
+    python -m venv .venv
+    ````
+2. Activate your virtual environment and install the backend dependencies.
+    ```bash
+    source .venv/bin/activate
+    pip install -e .
+    ```
+   
+### Frontend dependencies
+
+1. Install the frontend dependencies.
+    ```bash
+    cd web
+    yarn
+    ```
+
+## Running the server
+
+Run the full stack via Docker Compose:
+
+<!-- TODO: Consider adding `--build` to this command so that Docker Compose builds
+           the containers, rather than pulling from from GHCR (unless you
+           want to use the versions that happen to currently be on GHCR).
+           This has to do with the fact that the `docker-compose.yml` file 
+           contains service specs having both an `image` and `build` section. -->
```

Comment:
This comment came about as Eric and I were helping Yan get started with `nmdc-server` we both realized we were slightly hesitant if we completely understood the behavior of having both `build` and `image` in the definition of a Docker Compose service. But yeah, I think what you said is what I would expect.

Reply:
One approach here may be to add a reader-facing note to the instructions that explains the situation. Something like, "because `docker-compose.yml` contains both X and Y, when you run Z, Docker will do N. You can include the `--build` option to force docker to do M instead." Then, we could get rid of this HTML comment `<!-- ... -->` about adding `--build` to the command—we would be leaving it up to the reader.